### PR TITLE
runtime-rs: add parameter for propagation of (u)mount events

### PR DESCRIPTION
**Pull request by ananos**
Add an extra parameter in `bind_mount_unchecked` to specify the propagation type: "shared" or "slave".

Fixes: #7017

### CODE REVIEW

File: src/libs/kata-sys-util/src/mount.rs [hunk 1]
```diff
@@ -269,7 +270,12 @@ pub fn bind_mount_unchecked<S: AsRef<Path>, D: AsRef<Path>>(
     .map_err(|e| Error::BindMount(abs_src, dst.to_path_buf(), e))?;
 
     // Change into slave propagation mode.
-    mount(Some(""), dst, Some(""), MsFlags::MS_SLAVE, Some(""))
+    let propflag = match propagation {
```
Comment (Ji-Xinyou) [inline src/libs/kata-sys-util/src/mount.rs:273]: The function's comment doc gives
```text
Bind mount `src` to `dst` in slave mode, optionally in readonly mode if `readonly` is true.
```

Can you specify the need for this change? If the change is indeed needed, please change the comment doc on line 230.

**Review by Ji-Xinyou — APPROVED** [review]
LGTM, thanks for the work.

File: src/libs/kata-sys-util/src/mount.rs [hunk 2]
```diff
@@ -269,7 +270,12 @@ pub fn bind_mount_unchecked<S: AsRef<Path>, D: AsRef<Path>>(
     .map_err(|e| Error::BindMount(abs_src, dst.to_path_buf(), e))?;
 
     // Change into slave propagation mode.
```
Comment (fidencio) [inline src/libs/kata-sys-util/src/mount.rs:272]: s/into slave/into the chosen/

**Review by fidencio — APPROVED** [review]
@ananos, thanks for the PR.

I've left one comment that's a nit and I will leave it to you to address it or not.

lgtm!

File: src/libs/kata-sys-util/src/mount.rs [hunk 3]
```diff
@@ -828,7 +834,7 @@ mod tests {
             Err(Error::InvalidPath(_))
         ));
 
-        bind_mount_unchecked(tmpdir2.path(), tmpdir.path(), true).unwrap();
+        bind_mount_unchecked(tmpdir2.path(), tmpdir.path(), true, "slave").unwrap();
```
Comment (Tim-0731-Hzt) [inline src/libs/kata-sys-util/src/mount.rs:837]: I think you could just pass `MsFlags::MS_SLAVE ` as parameter, so you don't need to match the flag type in `bind_mount_unchecked`

Comment (Ji-Xinyou) [reply src/libs/kata-sys-util/src/mount.rs:837]: I agree, since it is not parsed from a configuration.

Comment (ananos) [reply src/libs/kata-sys-util/src/mount.rs]: sure, fixed!

**Review by Tim-0731-Hzt — APPROVED** [review]

**Review by studychao — APPROVED** [review]
LGTM, thanks!

File: src/libs/kata-sys-util/src/mount.rs [hunk 4]
```diff
@@ -227,7 +229,7 @@ pub fn bind_remount<P: AsRef<Path>>(dst: P, readonly: bool) -> Result<()> {
     do_rebind_mount(dst, readonly, MsFlags::empty())
 }
 
-/// Bind mount `src` to `dst` in slave mode, optionally in readonly mode if `readonly` is true.
+/// Bind mount `src` to `dst` with a custom propagation type, optionally in readonly mode if `readonly` is true.
 ///
```
Comment (Apokleos) [inline src/libs/kata-sys-util/src/mount.rs:233]: Thx.
Only two propagation types found. I think,  add a comment about types supported may be better.

**Pull request merged**